types/datetime.go

```diff
@@ -0,0 +1,302 @@
+package types
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+	"unicode"
+)
+
+// Datetime represents a Cedar datetime value
+type Datetime struct {
+	// Value is a timestamp in milliseconds
+	Value int64
```

Comment:
I think we should strongly consider keeping `Value` private and making an accessor method that gives you the time since the epoch in milliseconds.

Reply:
Changed. Note that Decimal still exposes this.

Reply:
I think Decimal will take a bit more effort, I've made an issue in the IDX project for us to handle that in a separate PR.